### ISSUE TITLE
Auth middleware is using "signIn" method on server side

### ISSUE THREAD

**Issue by timAdrien** [sidebase/nuxt-auth#537]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.16.1`
- Nuxt Version:     `3.7.4`
- CLI Version:      `3.9.0`
- Nitro Version:    `2.6.3`
- Package Manager:  `yarn@1.22.19`
- Builder:          `-`
- User Config:      `devtools`, `experimental`, `app`, `modules`, `runtimeConfig`, `auth`, `build`, `vite`, `i18n`, `eslint`, `viewport`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-rc.4`, `@nuxtjs/eslint-module@4.1.0`, `@sidebase/nuxt-auth@0.6.0-beta.6`, `@pinia/nuxt@0.4.11`, `@unlazy/nuxt@0.9.5`, `nuxt-viewport@2.0.6`, `vuetify-nuxt-module@0.5.12`
- Build Modules:    `-`

### Describe the bug

I'm using:

```
  auth: {
    provider: {
      type: 'authjs'
    },
    globalAppMiddleware: true
  }
```
in nuxt.config.ts

Which means that it enables the default auth middleware, here is the default implementation:

```
import { navigateTo, defineNuxtRouteMiddleware, useRuntimeConfig } from "#app";
import { determineCallbackUrl } from "../utils/url.mjs";
import { useAuth } from "#imports";
export default defineNuxtRouteMiddleware((to) => {
 const metaAuth = typeof to.meta.auth === "object" ? {
   unauthenticatedOnly: true,
   ...to.meta.auth
 } : to.meta.auth;
 if (metaAuth === false) {
   return;
 }
 const authConfig = useRuntimeConfig().public.auth;
 const { status, signIn } = useAuth();
 const isGuestMode = typeof metaAuth === "object" && metaAuth.unauthenticatedOnly;
 if (isGuestMode && status.value === "unauthenticated") {
   return;
 }
 if (typeof metaAuth === "object" && !metaAuth.unauthenticatedOnly) {
   return;
 }
 if (status.value === "authenticated") {
   if (isGuestMode) {
     return navigateTo(metaAuth.navigateAuthenticatedTo ?? "/");
   }
   return;
 }
 if (authConfig.globalAppMiddleware.allow404WithoutAuth) {
   const matchedRoute = to.matched.length > 0;
   if (!matchedRoute) {
     return;
   }
 }
 if (authConfig.provider.type === "authjs") {
   const signInOptions = { error: "SessionRequired", callbackUrl: determineCallbackUrl(authConfig, () => to.path) };
   return signIn(void 0, signInOptions);
 } else if (typeof metaAuth === "object" && metaAuth.navigateUnauthenticatedTo) {
   return navigateTo(metaAuth.navigateUnauthenticatedTo);
 } else {
   return navigateTo(authConfig.provider.pages.login);
 }
});
```

But as you can see there is no "process.server" check before running "signIn" method. And as shown in documentation: https://next-auth.js.org/getting-started/client#signin

this method is client side only.

This causes this error: [nuxt] [request error] [unhandled] [500] Cannot set headers after they are sent to the client

**Comment by Tcrussell436** [comment]
Looking at all these unanswered issues and the absolute cluster it is to set this up with a provider other than FB/Google/Github such as this here, defaultProvider not working and still redirecting to signin page, signOut "logs out" the user then redirects them to signIn where they are immediately signed again. Just.... lol... take a few weeks and just write your own and save the headache.

**Comment by zoey-kaiser** [comment]
Hi @timAdrien 👋 

Thanks for pointing this out. This will fix an annoying error, that NuxtAuth had! I created a fix in #644 🥳

**Comment by zoey-kaiser** [comment]
> Looking at all these unanswered issues and the absolute cluster it is to set this up with a provider other than FB/Google/Github such as this here, defaultProvider not working and still redirecting to signin page, signOut "logs out" the user then redirects them to signIn where they are immediately signed again. Just.... lol... take a few weeks and just write your own and save the headache.

@Tcrussell436 If you prefer writing your own integration, you can definitely do this! I am currently maintaining NuxtAuth in my free time, resulting in me not always being able to be on top of everything. But IMO this is part of Open Source. 
We just wanted to help the community by sharing how we handled authentication and do our best to maintain it in our free time. 
There are many alternatives to handling Auth, such as writing your own or using an alternative package (one being https://github.com/becem-gharbi/nuxt-auth, which I think is super well written and I whole heartily recommend looking into). Developers can then choose which option they prefer. If our Auth package does not meet your needs, that it completely fine!
